python/ray/serve/_private/router.py

```diff
@@ -449,6 +449,13 @@ async def _resolve_deployment_responses(
                     )
                 elif isinstance(obj, DeploymentResponse):
                     responses.append(obj)
+                    if obj not in request_args and obj not in request_kwargs.values():
```

Comment:
I assume you've tested that this check works and we don't spuriously print?

Reply:
Yup, this will only print for deployment response objects that weren't passed in as top level args/kwargs.